File: OgreWrapper/src/ConfigOption.cpp
#include "StdAfx.h"
#include "..\include\ConfigOption.h"
#include "MarshalUtils.h"

namespace OgreWrapper
{

ConfigOption::ConfigOption(Ogre::ConfigOption* configOption)
:possibleValues(gcnew System::Collections::Generic::List<System::String^>())
{
	this->name = MarshalUtils::convertString(configOption->name);
	this->currentValue = MarshalUtils::convertString(configOption->currentValue);
	Ogre::StringVector::iterator iter;
	for(iter = configOption->possibleValues.begin(); iter != configOption->possibleValues.end(); ++iter)
	{
		possibleValues->Add(MarshalUtils::convertString(*iter));
	}
	immutable = configOption->immutable;
}

System::String^ ConfigOption::Name::get() 
{
	return name;
}

System::String^ ConfigOption::CurrentValue::get() 
{
	return currentValue;
}

System::Collections::Generic::List<System::String^>^ ConfigOption::PossibleValues::get() 
{
	return possibleValues;
}

bool ConfigOption::Immutable::get() 
{
	return immutable;
}

}

File: OgrePlugin/Wrapper/Renderer/ConfigOption.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Runtime.InteropServices;

namespace OgrePlugin
{
    public class ConfigOption
    {
        private List<String> possibleValues = new List<string>();

        internal ConfigOption()
        {
            
        }

        public String Name { get; internal set; }

        public String CurrentValue { get; set; }

        public IEnumerable<String> PossibleValues
        {
            get
            {
                return possibleValues;
            }
        }

        public bool Immutable { get; internal set; }

        internal void _setDetails(IntPtr name, IntPtr currentValue, bool immutable)
        {
            this.Name = Marshal.PtrToStringAnsi(name);
            this.CurrentValue = Marshal.PtrToStringAnsi(currentValue);
            Immutable = immutable;
        }

        internal void _addPossibleValue(IntPtr possibleValue)
        {
            possibleValues.Add(Marshal.PtrToStringAnsi(possibleValue));
        }
    }
}
